cpp/src/arrow/engine/substrait/relation_internal.h

```diff
@@ -40,9 +40,46 @@ struct DeclarationInfo {
   int num_columns;
 };
 
+/// \brief A function to extract Acero Declaration from a Substrait Rel object
 ARROW_ENGINE_EXPORT
 Result<DeclarationInfo> FromProto(const substrait::Rel&, const ExtensionSet&,
                                   const ConversionOptions&);
 
+/// \brief Serializes a Declaration, produce a Substrait Rel and update the global
+/// Substrait plan. A Substrait Rel is passed as a the plan and it is updated with
+/// corresponding Declaration passed for serialization.
+///
+/// Note that this is a rather a helper method useful to fuse a partially serialized
+/// plan with another plan. The reason for having a partially serialized plan is to
+/// avoid unnecessary complication and enable partial plan serialization without
+/// affecting a global plan. Since kept as unique_ptr resources are relased efficiently
+/// upon releasing for the global plan.
```

Comment:
```suggestion
/// \brief Convert a Declaration (and its inputs) to a Substrait Rel
///
/// A Substrait Rel is passed as a the plan and it is updated with
/// corresponding Declaration passed for serialization.
///
/// Note that this is a rather a helper method useful to fuse a partially serialized
/// plan with another plan. The reason for having a partially serialized plan is to
/// avoid unnecessary complication and enable partial plan serialization without
/// affecting a global plan. Since kept as unique_ptr resources are relased efficiently
/// upon releasing for the global plan.
```

If this is a helper method does it need to appear in the `.h` file?

Reply:
Actually not a helper, but this interface is required for both `plan` and `relation` serialization.

Reply:
Updated the description.

Reply:
Probably remove this and only use internally is better as you suggested.